src/xmipp/libraries/reconstruction/subtract_projection.cpp

```diff
@@ -357,6 +363,154 @@ Matrix1D<double> ProgSubtractProjection::checkBestModel(MultidimArray< std::comp
 	return R2;
 }
 
+void ProgSubtractProjection::generateNoiseEstimationSideInfo()
+{
+	// Initialize powerNoise
+	powerNoise.initZeros((int)Ydim, (int)Xdim/2 +1);
+
+	// Calculate boundaries for noise estimation 
+	// (make a more eficinet sampling of the subtracted region)
+	if(maskVolProvided)
+	{
+		int minX = Xdim;
+		int minY = Ydim;
+		int minZ = Zdim;
+		int maxX = 0;
+		int maxY = 0;
+		int maxZ = 0;
+
+		for(size_t k=0; k<Zdim; ++k)
+		{
+			for(size_t i=0; i<Ydim; ++i)
+			{
+				for(size_t j=0; j<Xdim; ++j)
+				{
+					if (DIRECT_A3D_ELEM(maskVol(), k, i, j) > 0) {
```

Comment:
I would use DIRECT_MULTIDIM_ELEM(maskVol, n) with a counter ++n. Thus, the access is faster:
long n = 0;
for(size_t k=0; k<Zdim; ++k)
		{
			for(size_t i=0; i<Ydim; ++i)
			{
				for(size_t j=0; j<Xdim; ++j)
				{
					if (DIRECT_A3D_ELEM(maskVol(), n) > 0) 
						minX = std::min(minX, (int)i);
						minY = std::min(minY, (int)j);
						minZ = std::min(minZ, (int)k);
						maxX = std::max(maxX, (int)i);
						maxY = std::max(maxY, (int)j);
						maxZ = std::max(maxZ, (int)k);
					}
                                        ++n;
				}
			}
		}

Reply:
I was not aware of this, done.